### ISSUE TITLE
I was curious about the paper, I did a little bit comparison with Adam and AdamW and RAG optimizers on supervisod learning.

### ISSUE THREAD

**Issue by donymorph** [TobiasLv/RAD#2]
on mnist dataset there is no difference at all. I said ok.
let me try with time series dataset with lstm and transformer architectures 
down below is the tensorboard comparison graph.
<img width="1766" height="700" alt="Image" src="https://github.com/user-attachments/assets/b0642fd0-a959-44a1-9684-ffad88481798" />
""""""""""""""""""""""""
I also did with my own dataset.  it totally failed. stuck on local minima the same as SGD. 

<img width="1848" height="853" alt="Image" src="https://github.com/user-attachments/assets/078a1f97-a839-4d98-9538-61dddd0399f2" />

**Comment by TobiasLv** [comment]
Thank you for your interest in our work.

The RAD optimizer was originally proposed to mitigate gradient oscillations caused by stochastic sample distribution perturbations in reinforcement learning, with the goal of improving training stability and final policy performance. However, since its derivation relies only on the general formulation of stochastic non-convex optimization in neural networks, RAD is also applicable to supervised learning tasks.

That said, when applying RAD to supervised learning, particular attention should be paid to the tuning of two hyperparameters:

1. **`max_iter`**
   We recommend setting `max_iter` to match the actual number of network update steps. In supervised learning, training progress is often measured in epochs, but note that the optimizer operates on iterations. Therefore, a common setting is `max_iter = max_epoch / mini_batch_size`.

2. **`zeta`**
   The parameter `zeta` appears in the denominator of the effective learning rate. It increases from 0 to 1 over the course of training and is designed to suppress abnormal gradients as optimization proceeds. In supervised learning, however, the training dataset is typically fixed and the sample distribution is relatively stable. As a result, an overly aggressive growth of `zeta` may prematurely dampen learning dynamics and hinder rapid performance improvement in early training stages. Therefore, the growth schedule of `zeta` requires careful tuning.

   In `optim.py` (line 122), we provide a commented example configuration tailored for supervised learning tasks. Nevertheless, this setting is not universal and should be further adjusted based on task-specific understanding and empirical behavior.

Again, we appreciate your attention and feedback. We are also aware of the sensitivity of RAD to hyperparameter tuning in supervised settings, and we are currently exploring improved variants of RAD that can adaptively adjust these hyperparameters based on gradient dynamics, with a particular focus on large-model fine-tuning scenarios. Please stay tuned for future updates.
